akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -32,7 +32,7 @@ private[akka] object RemoteWatcher {
   final case class UnwatchRemote(watchee: InternalActorRef, watcher: InternalActorRef)
 
   @SerialVersionUID(1L) case object Heartbeat extends HeartbeatMessage
-  @SerialVersionUID(1L) final case class HeartbeatRsp(addressUid: Int) extends HeartbeatMessage
+  @SerialVersionUID(1L) final case class HeartbeatRsp(addressUid: Long) extends HeartbeatMessage
```

Comment:
This is so weird, even with a different serialuid, watching an actor works fine, expected heartbeating to fail with serialization error. Is there a difference between watch when using cluster and plain remoting (I tested with cluster)?